### ISSUE TITLE
Allow multiple chains in method_bayes()

### ISSUE THREAD

**Issue by tobiasmuetze** [insightsengineering/rbmi#417]
Currently, `method_bayes()` only runs a single Markov chain. Among other things, this limits the diagnostics that can performed. 

It would be great if the number of Markov chains to be run can be specified as input argument in `method_bayes()`. Ideally, it should also be allowed to run them in parallel through the `cores` argument.

**Comment by nociale** [comment]
@gowerc I also cannot remember any reason other than the warmup phase which will have to be repeated for each chain. I agree that allowing the user to choose the number of chains (as an optional argument) can be an enhancement for the package.

**Comment by nociale** [comment]
@gowerc True, we would need to add some randomness to the initial values of the chains. Another (simpler) option to the one you suggest could be to just use the rstan default option for the initial values (https://mc-stan.org/docs/2_25/reference-manual/initialization.html). We may need to increase the default warmup phase in this case with respect to using only 1 chain initialized at the REML parameter estimates from a MMRM model. The user could then override the default option of using one chain and choose "X" chains with random initial values. This option could be potentially explored as it should be relatively easy to implement (?). In my personal experience with our implementation of Bayesian MI, I could not find cases of clear convergence issues, so I would agree that this could be considered a "nice to have" more than a "must have" for the package. 

@wolbersm do you have any thought about the relevance of adding an option to allow the user to run multiple chains for the MCMC step of Bayesian MI?

**Comment by wolbersm** [comment]
@nociale No, not really. In principle, I am supportive but also I agree that this is not a "must have". As you write, currently, the chain is started from "good" starting values (the REML parameter estimates). Before multiple chains are supported, it would be good to make some experiments to ensure that chains converge to the stationary distribution within reasonable time frame if started from random values.

**Comment by luwidmer** [comment]
See #472 for the design doc / discussion for implementing this
